pkg/logs/client/http/destination.go

```diff
@@ -295,8 +297,13 @@ func (d *Destination) unconditionalSend(payload *message.Payload) (err error) {
 	if err != nil {
 		// the read failed because the server closed or terminated the connection
 		// *after* serving the request.
+		log.Debugf("Server closed or terminated the connection after serving the request with err %v", err)
 		return err
 	}
+
+	metrics.DestinationHttpRespByStatus.Add(strconv.Itoa(resp.StatusCode), 1)
+	metrics.TlmDestinationHttpRespByStatus.Inc(strconv.Itoa(resp.StatusCode))
```

Comment:
will we know which endpoint is failing to send by recording this? 


Also, should we be recording _something_ when we [hit network issues](https://github.com/DataDog/datadog-agent/pull/15058/files#diff-9e74f15ec100d26628739bfd3f6dd33d24fda6af1c843993a21b268ec97b4c1eR287-R292) aka the context gets canceled & we retry?

Reply:
i can add the url as a tag to this metric, but we don't have clean endpoint or pipeline names at this point. 

for some reason this link just drops me into the page of all files changed, are you pointing to specific code location?

Reply:
Oh sorry, yeah the context cancellation is being caught above this code & then we re-try so im wondering if we want to record these timeouts happening

Reply:
ah we actually do record those already when the error is returned in sendAndRetry: https://github.com/DataDog/datadog-agent/blob/master/pkg/logs/client/http/destination.go#L218-L224

Reply:
Can you show me where the metrics are that record the timeout?? Didn't know that existed